### PR TITLE
DEVPROD-5318: update unattainable dependencies in bulk

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-5318

### Description
[An example HC trace](https://ui.honeycomb.io/mongodb-4b/environments/production/datasets/evergreen/result/cT2VBKpd4CE/trace/aKkt735hVpG?fields[]=s_name&fields[]=s_serviceName&span=7a2f6ea93711e813) (and other examples I looked through) suggest that the slowest part of end task is marking dependencies unattainable after a task fails that has lots of dependencies. For example, if one of the compile tasks in a server patch fails, it spends all of end task blocking thousands of tasks that depend on compile before the request eventually times out.

This attempts to optimize it by reducing the number of round-trips to the DB. It's the same update, but instead of doing thousands of single updates, it'll instead do one big bulk update. I'm not that confident that this will improve the performance enough, so I'll be monitoring the impact after this is deployed.

* Block/unblock tasks with unattainable dependencies in bulk.
* Log task blocked events in bulk.
* Refactor tests a little bit.

### Testing
* Existing tests cover that functionality works like before.
* Added a unit test for updating many tasks at once.
* Ran patches in staging to verify that it blocked dependencies like it did before.

### Documentation
N/A

### CODE REVIEW

File: model/event/task_event.go [hunk 1]
```diff
@@ -164,6 +164,44 @@ func LogTaskRestarted(taskId string, execution int, userId string) {
...
+				BlockedOn: d.BlockedOn,
+			},
+		}
+		events = append(events, e)
+	}
+	if err := LogManyEvents(events); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"resource_type": ResourceTypeTask,
+			"event_type":    TaskBlocked,
+			"message":       "error logging event",
```
Comment (ablack12) [inline model/event/task_event.go:199]: error logging events probably

File: model/task/task.go [hunk 2]
```diff
@@ -2645,19 +2651,44 @@ func (t *Task) MarkUnscheduled() error {
...
+// returns all the tasks after the update.
+func MarkAllForUnattainableDependency(ctx context.Context, tasks []Task, dependencyID string, unattainable bool) ([]Task, error) {
+	if len(tasks) == 0 {
+		return tasks, nil
+	}
+
+	taskIDs := make([]string, 0, len(tasks))
+	newlyBlockedTaskData := make([]event.TaskBlockedData, 0, len(tasks))
+	for _, t := range tasks {
+		if !t.Blocked() && unattainable && !t.OverrideDependencies {
```
Comment (ablack12) [inline model/task/task.go:2666]: why do we only want to log in this case? If we're overriding dependencies, should we also not be adding to taskIDs?

Comment (Kimchelly) [reply model/task/task.go:2666]: I don't really know why overriding dependencies matters, but this check is equivalent to the old one, so I just kept the logic the same. It should still log task blocked under the same conditions as before.

File: model/event/task_event.go (hunk 1 above)
Comment (Kimchelly) [reply model/event/task_event.go]: Fixed this one, along with all the other `LogManyEvents`.

**Review by ablack12 — APPROVED** [review]
🚀

**Pull request merged**